pymarsys/emarsys.py

```diff
@@ -0,0 +1,188 @@
+import asyncio
+import base64
+import datetime
+import hashlib
+import json
+import logging
+import os
+import uuid
+
+from aiohttp import ClientSession
+import requests
+
+from .contact import Contact
+
+VALID_HTTP_METHODS = ('GET', 'POST', 'PUT', 'DELETE')
+EMARSYS_URI = 'https://api.emarsys.net/'
+
+logger = logging.getLogger(__name__)
+
+
+class ApiCallError(Exception):
+    pass
+
+
+class Emarsys:
+    """
+    Make authenticated calls to Emarsys' API through its methods
+    Usage example:
+        >>> client = Emarsys('sync', <user>, <password>)
+        >>> client.settings
+        >>> {
+                'data': {
+                    'country': 'France',
+                    'environment': 'suite16.emarsys.net',
+                    'id': 123456789,
+                    'name': 'name',
+                    'password_history_queue_size': 1,
+                    'timezone': 'Europe/Vienna',
+                    'totalContacts': '1'
+                },
+                'replyCode': 0,
+                'replyText': 'OK'
+            }
+    """
+    def __init__(self, username, secret, uri=EMARSYS_URI, is_async=False):
+        emarsys_settings_endpoint = 'api/v2/settings'
+        self.is_async = is_async
+        self.username = username
+        self.secret = secret
+        self.uri = uri
+        if self.is_async:
+            self.make_call = self.__make_async_call
+            loop = asyncio.get_event_loop()
```

Comment:
i don't know the best practice here, but what if this async code is used in another loop ?
```
new_event_loop()
```